server/src/test/java/org/elasticsearch/index/mapper/MapperBuilderContextTests.java

```diff
@@ -27,4 +27,11 @@ public void testRootWithMergeReason() {
         assertEquals(mergeReason, root.getMergeReason());
     }
 
+    public void tesIsInNestedContext() {
+        MapperBuilderContext root = MapperBuilderContext.root(true, false);
```

Comment:
It doesn't affect the test, but why did you set `isSourceSynthetic` to `true` here? It doesn't seem to have any impact on the tested functionality.

Reply:
Correct, I just filled in with some data. I could use random values but didn't think it worthwhile, as it won't impact the response.

Reply:
Agreed, random values aren't worth it here. 99% of times I've used `MapperBuilderContext.root`, both args were set to `false`, so IMO that's good enough for here too.